vendor/engines/secure_rooms/app/services/secure_rooms/access_rules/account_selection_rule.rb

```diff
@@ -5,25 +5,24 @@ module AccessRules
     class AccountSelectionRule < BaseRule
 
       def evaluate
+        accounts = user.accounts_for_product(card_reader.secure_room)
+        selected_account = accounts.find { |account| account.id == requested_account_id }
+
         if accounts.blank?
-          deny! "User has no valid accounts for this Product"
+          deny!(reason: "User has no valid accounts for this Product")
         elsif selected_account.present?
-          grant!
+          grant!(accounts: accounts)
```

Comment:
I don't think so but I'm not sure... in theory the grant just lets the user in without need for them to see any new info at this point. I'd leave it until we have a firm answer for this and then it should be a quick-ish change